module_utils/xpath.py

```diff
@@ -288,6 +288,9 @@ def filterNodes(self, nodes):
                 selection.append(node)
 
         log.debug("Selector: candidates are %s" % [str(n) for n in selection])
+        # sort handles to get correct order
+        selection.sort(key=lambda n: str(n))
```

Comment:
I am wondering whether `n.handle` would be better than `str(n)`?

Reply:
Let me try it.